File: user-service/src/main/java/socialapp/userservice/repository/common/CommonRepository.java
package socialapp.userservice.repository.common;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.NoRepositoryBean;
import socialapp.userservice.common.exception.EntityNotFoundException;

import java.io.Serializable;

@NoRepositoryBean
public interface CommonRepository<E,P extends Serializable> extends JpaRepository<E,P> {

    default E findByID(P id){
        return findById(id)
                .orElseThrow(() -> new EntityNotFoundException(entityClass(),id));
    }

    Class<?> entityClass();
}

File: kt-user-service/src/main/kotlin/socialapp/ktuserservice/repository/common/CommonRepository.kt
package socialapp.ktuserservice.repository.common

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.NoRepositoryBean
import socialapp.ktuserservice.common.exception.EntityNotFoundException
import java.io.Serializable

@NoRepositoryBean
interface CommonRepository<E, P : Serializable> : JpaRepository<E, P> {

    fun findByID(id: P): E {
        return findById(id)
            .orElseThrow { EntityNotFoundException(entityClass(), id) }
    }

    fun entityClass(): Class<*>
}